helix-term/src/ui/editor.rs

```diff
@@ -856,6 +856,52 @@ impl Component for EditorView {
                 doc.set_selection(view.id, selection);
                 EventResult::Consumed(None)
             }
+
+            Event::Mouse(MouseEvent {
+                kind: MouseEventKind::ScrollUp | MouseEventKind::ScrollDown,
+                row,
+                column,
+                ..
+            }) => {
+                let editor = &mut cxt.editor;
+                let current_view = editor.tree.focus;
+
+                let direction = match event {
+                    Event::Mouse(MouseEvent {
+                        kind: MouseEventKind::ScrollUp,
+                        ..
+                    }) => editor.config.scroll_lines.signum(),
+                    Event::Mouse(MouseEvent {
+                        kind: MouseEventKind::ScrollDown,
+                        ..
+                    }) => -editor.config.scroll_lines.signum(),
+                    _ => 0,
+                };
+                let cmd = match direction {
+                    1 => commands::Command::scroll_up,
+                    -1 => commands::Command::scroll_down,
+                    _ => return EventResult::Ignored,
+                };
+
+                let result = editor.tree.views().find_map(|(view, _focus)| {
+                    view.pos_at_screen_coords(&editor.documents[view.doc], row, column)
+                        .map(|_| view.id)
+                });
+
+                match result {
+                    Some(view_id) => editor.tree.focus = view_id,
+                    None => return EventResult::Ignored,
+                }
+
+                for _ in 0..editor.config.scroll_lines.abs() {
+                    cmd.execute(&mut cxt);
+                }
```

Comment:
This is actually really inefficient, should use https://github.com/helix-editor/helix/blob/e977fbea89b3b549889f55bdf5c9764def2b64fd/helix-term/src/commands.rs#L874